common/beerocks/bcl/include/bcl/beerocks_socket_thread.h

```diff
@@ -69,6 +70,7 @@ class socket_thread : public thread_base {
 
     uint8_t rx_buffer[message::MESSAGE_BUFFER_LENGTH];
     uint8_t tx_buffer[message::MESSAGE_BUFFER_LENGTH];
+    uint8_t cert_tx_buffer[message::MESSAGE_BUFFER_LENGTH];
```

Comment:
It doesn't seem right to define it here. This is only relevant to the controller, and maybe will be needed to agent certification as well.
I think that it will be more correct to define it on the controller/backhaul since they are the only ones that use it, or at least define it on the BTL which the controller and the backhaul are derived from.

Anyway I would have defined only a pointer to the buffer and cert_tx_buffer and allocated or runtime id the ucc_listener is up (port!=0) or certification_mode=1.

Reply:
I could split it to the controller (son_master_thread) and agent (backhaul_manager_thread) but I think it is pointless and wasted effort since this is anyway a hack which we should remove eventually once we have a real channel selection task which will store all the needed parameters in the DB and then we will re-trigger channel selection and be done with it, as @arnout suggested multiple times.

So for now - I think there is no harm with keeping the certification stuff in the base class. Its even better since we'll have less code to remove once we go with the correct solution.